components/EventDetail.js

```diff
@@ -9,43 +14,173 @@ const StyledContainer = styled.div`
+      title: event.title,
+      city: event.city,
+      category: event.category,
+      startDateTime: event.startDateTime,
+      organizer: event.organizer,
+      description: event.description,
+    });
+  }, [event]);
+
+  // handle functions
+  async function handleSubmit(jsEvent) {
+    jsEvent.preventDefault();
+    await updateEvent(id, formData);
+    setIsEditMode(false);
+    mutate();
+  }
+  function handleChange(jsEvent, target) {
+    setFormData((prevFormData) => {
+      return {
+        ...prevFormData,
+        [target]: jsEvent.target.value,
+      };
+    });
+  }
   return (
-    <StyledContainer>
-      <StyledHeaderImage
-        src={event.imageUrl}
-        alt={event.title}
-        width={260}
-        height={260}
-      />
-
-      <StyledEventInfoContainer>
-        <StyledTitle>{event.title}</StyledTitle>
-        <StyledEventInfo $right>{event.category}</StyledEventInfo>
-        <StyledEventInfo>{event.city}</StyledEventInfo>
-        <StyledEventInfo
-          $right
-        >{`${day}. ${month} ${year}, ${time} Uhr`}</StyledEventInfo>
-        <StyledEventInfo>{event.organizer}</StyledEventInfo>
-      </StyledEventInfoContainer>
-      <StyledDescription>{event.description}</StyledDescription>
-    </StyledContainer>
+    <>
+      <StyledContainer>
+        <StyledHeaderImage
+          src={event.imageUrl}
+          alt={event.title}
+          width={260}
+          height={260}
+        />
+        <StyledEventInfoContainer>
+          {isEditMode ? (
+            <StyledTitle>
+              <label htmlFor="title">Title</label>
+              <input
+                id="title"
+                value={formData.title}
+                onChange={(jsEvent) => handleChange(jsEvent, "title")}
+              />
+            </StyledTitle>
+          ) : (
+            <StyledTitle>{event.title}</StyledTitle>
+          )}
+          {isEditMode ? (
+            <StyledEventInfo $right>
+              <label htmlFor="category">Category</label>
+              <input
+                id="category"
+                value={formData.category}
+                onChange={(jsEvent) => handleChange(jsEvent, "category")}
+              />
+            </StyledEventInfo>
+          ) : (
+            <StyledEventInfo $right>{event.category}</StyledEventInfo>
+          )}
+
+          {isEditMode ? (
+            <StyledEventInfo>
+              <label htmlFor="city">City</label>
+              <input
+                id="city"
+                value={formData.city}
+                onChange={(jsEvent) => handleChange(jsEvent, "city")}
+              />
+            </StyledEventInfo>
+          ) : (
+            <StyledEventInfo>{event.city}</StyledEventInfo>
+          )}
+
+          {isEditMode ? (
+            <StyledEventInfo $right>
+              <label htmlFor="date">Date</label>
+              <input
+                id="date"
+                type="datetime-local"
+                value={formData.date}
```

Comment:
When the user clicks on the edit-button all input elements are prefilled with data except of the date. I can see in your Event Schema that you are saving the date as type date in your database. The input element cannot read and display this format. You will need to transform the date, so that the input element can actually display the date. 

If you do not want to use the date for any kind of calculations, you can also save the date as a string in your database.

Reply:
Thank you very much. We implemented a caluclation in the next version :)